src/data/finset.lean

```diff
@@ -1357,6 +1357,37 @@ protected def subtype {α} (p : α → Prop) [decidable_pred p] (s : finset α)
   ∀{a : subtype p}, a ∈ s.subtype p ↔ a.val ∈ s
 | ⟨a, ha⟩ := by simp [finset.subtype, ha]
 
+/-- `s.subtype p` converts back to `s.filter p` with
+`embedding.subtype`. -/
+lemma subtype_map (p : α → Prop) [decidable_pred p] :
+  (s.subtype p).map (function.embedding.subtype _) = s.filter p :=
```

Comment:
`@[simp]`?

Reply:
Added more use of `@[simp]` as suggested.